lua/bookmarks/init.lua

```diff
@@ -31,6 +31,48 @@ M.goto_bookmark = function()
   end)
 end
 
+M.goto_next_bookmark = function()
+  local active_list = Repo.get_active_list()
+  local bookmarks = Node.get_all_bookmarks(active_list)
+  local filepath = vim.fn.expand("%:p")
+  local cur_lnr = vim.api.nvim_win_get_cursor(0)[1]
+
+  local selected_bm
+  for _, bookmark in ipairs(bookmarks) do
+    if filepath == bookmark.location.path and bookmark.location.line > cur_lnr then
+      if not selected_bm or bookmark.location.line < selected_bm.location.line then
+        selected_bm = bookmark
+      end
+    end
+  end
+
+  if selected_bm then
+    Service.goto_bookmark(selected_bm.id)
+    Sign.safe_refresh_signs()
+  end
+end
+
+M.goto_prev_bookmark = function()
```

Comment:
can you help to put those two method into the service layer?
Because I want keep the init.lua only care about call the api and manage user input

Reply:
Thanks for the review! I've moved the logic to the Service layer and refactored the code.

Reply:
> And you can just ignore the AI review comments

Oops, I didn't spot this until now. I did as was suggested for the code duplication refactoring.
Personally preferred just duplicated code, it was immediately readable and there is only two direction for the finding bookmarks for this feature.

Let me know if you want me to change it back, else we keep it as is :blush:

Reply:
Yeah I prefer that one as well~
Readability means a lot to me